weldx/welding/groove/iso_9692_1.py

```diff
@@ -99,16 +100,39 @@ def plot(
             title, raster_width, None, axis, axis_label, grid, line_style, ax=ax
         )
 
+    @abstractmethod
     def to_profile(self, width_default: pint.Quantity = None) -> geo.Profile:
         """Implement profile generation.
 
         Parameters
         ----------
-        width_default :
+        width_default : pint.Quantity['length']
              optional width to extend each side of the profile (Default value = None)
 
         """
-        raise NotImplementedError("to_profile() must be defined in subclass.")
+        pass
+
+    def compute_welding_speed(self, seam_length: pint.Quantity,
```

Comment:
I would suggest only defining an (abstract) `area` property/method on the groove types and moving the welding speed calculation to another module

```python
wire_area = np.pi / 4 * wire_diameter
weld_speed = wire_area * wire_feed / groove_area
```